src/indexer.rs

```diff
@@ -99,45 +107,64 @@ impl Indexer {
         self.inner.clone()
     }
 
+    pub async fn connect_to(&mut self, target: &str) -> Result<()> {
+        info!("Connecting to node for data availability stream");
+        let listener = TcpListener::bind("127.0.0.1:0")
+            .await
+            .expect("Failed to bind to port");
+
+        let addr = listener
+            .local_addr()
+            .expect("Failed to get local port")
+            .to_string();
+
+        let waiter = listener.accept();
+
+        Client::new()
+            .get(format!("{}v1/da/stream_request", target))
+            .query(&[("start_height", "1"), ("peer_ip", &addr)])
+            .timeout(std::time::Duration::from_secs(5))
+            .send()
+            .await?;
```

Comment:
I'm not convinced by the usage of a http request to do before creating the stream, I don't understand which problem it solves and I feel it adds complexity where it is not really required. It could be usefull if we consider that on a same node we could have undreds of indexer connected, but that's not the case. I think that in the high majority of case we will have 0 or 1 indexer connecter... then 1 single fixed port in the configuration is enough

Reply:
but that's not blocking, it's only my opinion :)

Reply:
Mostly I kind of liked not having several ports for different things, as that makes readability/documentation down the line harder to grok. It's not a huge deal, could be reverted.
I think it's going to be either 0, 1, or hundreds TBH, so I think this design makes some sense.

I also just wanted to write it out and see how it felt 😃.

Reply:
Ah mais je pense à une bonne raison de pas le faire: ça va pas marcher dans un contexte docker ou k8s, puisque le port ouvert par la node ne sera pas visible de l'extérieur pour envoyer des messages.

Reply:
Ça me semble probablement naturel parce que ça se fait dans un contexte "gaming multijoueur" où tu es pas forcément sûr du port en avance.